libraries/AP_Logger/AP_Logger_File.cpp

```diff
@@ -715,6 +709,39 @@ void AP_Logger_File::stop_logging(void)
     }
 }
 
+/*
+  wrapper for PrepForArming to move start_new_log to the logger thread
+ */
+void AP_Logger_File::PrepForArming()
+{
+#if APM_BUILD_TYPE(APM_BUILD_Replay)
+    AP_Logger_Backend::PrepForArming();
```

Comment:
We can't *not* call `PrepForArming` in the normal case without putting a warning on the base class method that it is completely overridden by the subclass.

It's probably better to break out a new `PrepForArming_start_new_log` virtual method and override that instead of overriding `PrepForArming`.  The base-class method just calling `start_new_log()`, the subclass overriding to do nothing and pointing ot where the open actually happens.

It also means that we're not duplicating any bugs around the rotate-pending stuff.

Reply:
it's a bit awkward with the FOREACH macro and the logging_started() check
I've broken out the rotate check, and added a note on the parent class